6e - deploy from GitHub - for issue #862 (#28)
nimbella/docs: docs/deploy-from-github.md

---
id: deploy-from-github
title: Deploy from GitHub
sidebar_label: Deploy from GitHub
---

import useBaseUrl from '@docusaurus/useBaseUrl';

Nimbella gives you the ability to deploy your applications and serverless functions that are on your GitHub.

> The process is very similiar to how you'd normally deploy your code. You start by running `nim project deploy` in your command prompt. But instead providing the local folder path, you provide the link for GitHub.

Reviewer: ```suggestion
The process is similar to how you normally deploy your code. You start by running `nim project deploy` in your command prompt. But instead providing the local folder path, you provide the link for GitHub.
```

> To test what it's like to deploy an app from GitHub, run this command to deploy the `visits` demo app.

Reviewer: Before giving this example, we should show what the format is for specifying the github repo. I think the existing nim docs cover this though: https://apigcp.nimbella.io/downloads/nim/nim.html#deploying-directly-from-github

---
id: deploy-from-github
title: Deploy from GitHub
sidebar_label: Deploy from GitHub
---

import useBaseUrl from '@docusaurus/useBaseUrl';

Nimbella gives you the ability to deploy your applications and serverless functions that are on your GitHub.

To indicate you want to deploy from GitHub, use a project path that starts with one of the following.

```
github:
git@github.com:
https://github.com/
```

The deployer supports all three prefix styles to align with developer habits and URLs copied from elsewhere: all three are equivalent and authenticate to GitHub in the same way.

### Specifying what you want from GitHub

The process is similar to how you normally deploy your code. You start by running `nim project deploy` in your command prompt. But instead of providing the local folder path, you provide the link for GitHub.

You follow the prefix with the “owner” (GitHub account), repository, the path to the project within the repository (if any), and specific branch or commit (if not `master`). For example,

```

> nim project deploy github:nimbella/demo-projects/visits

Reviewer: ```suggestion
```

nim project deploy git@github.com:/my-account/my-repo-with-project/#dev
```

> To test what it's like to deploy an app from GitHub, run this command to deploy the `visits` demo app.

Reviewer: ```suggestion
To test what it's like to deploy an app from GitHub, run this command to deploy the [Page Visits demo app](https://github.com/nimbella/demo-projects/tree/master/visits).
```
